### ISSUE TITLE
HallSubgroups, HallSystem, SyloSubgroups seem to work independently

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-system/gap#511]
A similar issue was already raised in #412 

If one computes a HallSystem, the computed Hall subgroups will not be added to ComputedHallSubgroups. Further, if SylowSubgroups are computed, those are not added to ComputedHallSubgroups. However, if a HallSubgroup is computed for a set of only one prime, then this Hall subgroup (in fact a Sylow subgroup) will be added to ComputedSylowSubgroups, as well. 

```
gap> G := DirectProduct(DihedralGroup(8), SmallGroup(27, 3));
<pc group of size 216 with 6 generators>
gap> HallSystem(G); 
[ Group([  ]), Group([ f1, f2, f3 ]), Group([ f1, f2, f4, f5, f3, f6 ]), 
  Group([ f4, f5, f6 ]) ]
gap> List(HallSystem(G), Size);
[ 1, 8, 216, 27 ]
gap> ComputedHallSubgroups(G); 
[  ]
gap> ComputedSylowSubgroups(G); 
[  ]
gap> SylowSubgroup(G, 2);
Group([ f1, f2, f3 ])
gap> ComputedSylowSubgroups(G); 
[ 2, Group([ f1, f2, f3 ]) ]
gap> ComputedHallSubgroups(G);
[  ]
gap> HallSubgroup(G, [3]);
Group([ f4, f5, f6 ])
gap> ComputedHallSubgroups(G); 
[ [ 3 ], Group([ f4, f5, f6 ]) ]
gap> ComputedSylowSubgroups(G); 
[ 2, Group([ f1, f2, f3 ]), 3, Group([ f4, f5, f6 ]) ]
```

I understand that in some sense it is a waste of space if a Sylow subgroup is stored in two lists. Would it be possible to maybe link these two values to both point to the same place in memory? 

And of course, if a HallSystem (or SylowSystem for that matter) is computed, then these really should be added to the corresponding ComputedHallSubgroups and ComputedSylowSubgroups lists. How to add them is another matter, though, because the odd entries of these lists are sorted.

**Comment by bh11** [comment]
In #514, I've just suggested tester and setter functions for these cases. In any case, you can have a look at the code to see how to insert a key/value pair.

**Comment by alex-konovalov** [comment]
Has this been resolved by #535 as well as #412?

**Comment by hungaborhorvath** [comment]
Yes, it has.
